### PR TITLE
chore(jest): remove examples before running tests

### PR DESCRIPTION
**Pull request by SimenB**
See https://github.com/nodejs/citgm/pull/867#issuecomment-860657374

##### Checklist

- [ ] `npm test` passes
- [ ] tests are included
- [ ] documentation is changed or added
- [x] contribution guidelines followed
      [here](https://github.com/nodejs/citgm/blob/HEAD/CONTRIBUTING.md)

### CODE REVIEW

File: lib/lookup.json [hunk 1]
```diff
@@ -253,7 +253,8 @@
     "prefix": "v",
     "maintainers": ["cpojer", "scotthovestadt", "SimenB", "thymikee", "jeysal"],
     "yarn": true,
-    "scripts": ["build:js", "test-ci-partial"],
+    "head": true,
```
Comment (SimenB) [inline lib/lookup.json:256]: https://github.com/facebook/jest/commit/5f7ae2117c502ef42695303d5f96d4e1ccb9ad22 hasn't been released

**Review by richardlau — APPROVED** [review]

**Review by targos — APPROVED** [review]

**Comment by targos** [comment]
CI: https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker-pipeline/147/console

**Comment by targos** [comment]
This introduces another error: 

```
15:33:08 error:                     | [31mError: Can't find a root directory while resolving a config file path.[39m                                                                                                                                                              
15:33:08 error:                     | [31mProvided path to resolve: /home/iojs/tmp/citgm_tmp/8843d9b7-75d1-4c8b-92ff-b4fe91cfc8a6/jest/examples/*[39m                                                                                                                             
15:33:08 error:                     | [31mcwd: /home/iojs/tmp/citgm_tmp/8843d9b7-75d1-4c8b-92ff-b4fe91cfc8a6/jest[39m                                                                                                                                                             
15:33:08 error:                     | [31m    at _default (/home/iojs/tmp/citgm_tmp/8843d9b7-75d1-4c8b-92ff-b4fe91cfc8a6/jest/packages/jest-config/build/resolveConfigPath.js:106:11)[39m                                                                                         
15:33:08 error:                     | [31m    at readConfig (/home/iojs/tmp/citgm_tmp/8843d9b7-75d1-4c8b-92ff-b4fe91cfc8a6/jest/packages/jest-config/build/index.js:223:49)[39m                                                                                                   
15:33:08 error:                     | [31m    at /home/iojs/tmp/citgm_tmp/8843d9b7-75d1-4c8b-92ff-b4fe91cfc8a6/jest/packages/jest-config/build/index.js:454:18[39m                                                                                                                
15:33:08 error:                     | [31m    at Array.map (<anonymous>)[39m                                                                                                                                                                                                      
15:33:08 error:                     | [31m    at readConfigs (/home/iojs/tmp/citgm_tmp/8843d9b7-75d1-4c8b-92ff-b4fe91cfc8a6/jest/packages/jest-config/build/index.js:448:10)[39m                                                                                                  
15:33:08 error:                     | [31m    at async runCLI (/home/iojs/tmp/citgm_tmp/8843d9b7-75d1-4c8b-92ff-b4fe91cfc8a6/jest/packages/jest-core/build/cli/index.js:220:59)[39m                                                                                               
15:33:08 error:                     | [31m    at async Object.run (/home/iojs/tmp/citgm_tmp/8843d9b7-75d1-4c8b-92ff-b4fe91cfc8a6/jest/packages/jest-cli/build/cli/index.js:163:37)[39m    
```

**Comment by SimenB** [comment]
@targos hah, I should have tested it 😅 Pushed a new commit which works locally 👍

**Comment by targos** [comment]
rebuild: https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker-pipeline/148/console

**Comment by targos** [comment]
There are errors related to "duplicate manual mocks": https://ci.nodejs.org/job/citgm-smoker-nobuild/1093/nodes=fedora-last-latest-x64/console

**Comment by SimenB** [comment]
those are just warnings, shouldn't affect the test run. Or does CITGM fail if there is output to stderr?

**Comment by targos** [comment]
> does CITGM fail if there is output to stderr

No, CITGM fails if the command returned a non-zero exit code.

**Comment by targos** [comment]
You can find the logs here: https://ci.nodejs.org/job/citgm-smoker-nobuild/1093/nodes=fedora-last-latest-x64/console

**Comment by SimenB** [comment]
I don't see any test failures in that log, tho... I assume `The canary is dead` just means it failed and not a timeout?

**Comment by targos** [comment]
With the last commit: `Usage Error: Couldn't find a script named "jest --color -i"`

**Comment by targos** [comment]
I tried with the previous commit on my mac and it fails with a bunch of errors like this:

```
error:                     |   ● transitive dependencies › does not mock deep dependencies when using deepUnmock
error:                     |
error:                     | ENAMETOOLONG: name too long, open '/private/var/folders/vy/7hd71t752szdg3km0llpsqdw0000gn/T/17b00a38-fb3b-4431-b5
error:                     |
error:                     |       at Object.writeFileSync (packages/jest-serializer/build/index.js:100:15)
error:                     |
error:                     | ...........................
```

**Comment by targos** [comment]
I can reproduce the failure on Linux.

**Comment by targos** [comment]
I don't know if it's why it fails on our CI, but locally I have quite a lot of errors. I did:

```console
wget https://github.com/facebook/jest/archive/HEAD.tar.gz
tar xf HEAD.tar.gz
cd jest-91b94c812834753adced4f1a8be0e6fc8937bb1f
yarn
yarn remove-examples
yarn build:js
export CI=true
yarn test-ci-partial
```

Output: https://gist.github.com/targos/d4bffbe8b51a7b2af5eb6b6200580f1f

**Comment by SimenB** [comment]
We're failing on CI using node 16.3.0 ourselves, on both mac and linux. Windows is passing. Not sure if it's the same failures or not, but it looks like the things we see on CI here (GH Actions times it out after 6 hours)

**Comment by SimenB** [comment]
@targos could you retry now 16.4.0 is out?

**Comment by targos** [comment]
First try with GitHub actions: https://github.com/nodejs/citgm/actions/runs/980018267

**Comment by targos** [comment]
CI: https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker-pipeline/151/

**Comment by targos** [comment]
The only error in CI is related to a full disk on one machine, merging!

**Comment by SimenB** [comment]
Wonderful! Hopefully this can actually pick up all the regressions in `async_hooks` in the future 😅 Are CITGM not run before v16 releases? Or didn't a failing suite raise any flags due to the mongodb errors?

**Comment by targos** [comment]
CITGM is always runs before a release. I suppose the mongodb errors prevented us from seeing the real issue.

**Pull request merged**